tests/core/test_listeners.py

```diff
@@ -108,6 +108,264 @@ def test_callback_is_not_called_when_resource_is_filtered(self):
 
         self.assertFalse(self.demo_mocked.return_value.called)
 
+    # uri field is blank in utils.view_lookup
+    # and there us no upath_info field in request
+    @mock.patch('kinto.core.utils.parse_resource',
+                return_value={'bucket': 'bid',
+                              'collection': 'cid'})
+    @mock.patch('kinto.core.utils.view_lookup',
+                return_value=('record',
+                              {'bucket_id': 'bid', 'collection_id': 'cid'}))
```

Comment:
There's a `DummyRequest` class in kinto.core.testing that provides at least `upath_info`. Do we still need these mocks if we can use that instead? I'd rather test against a DummyRequest but with real utils methods than a real Request but with mocked utils methods.